iac-extensions/arm/src/test/resources/checks/LogRetentionCheck/Microsoft.Network_firewallPolicies/test.json

```diff
@@ -0,0 +1,88 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "resources": [
+    {
+      "name": "Raise Issue as retentionDays are too low",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "isEnabled": true,
+          "retentionDays": 7
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as retentionDays are missing",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "isEnabled": true
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as isEnabled is missing",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "retentionDays": 15
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as retentionDays are too low and isEnabled is false",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "isEnabled": false,
+          "retentionDays": 7
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as isEnabled is false",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "isEnabled": false,
+          "retentionDays": 15
+        }
+      }
+    },
+    {
+      "name": "Raise Issue as insights is missing",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+      }
+    },
+    {
+      "name": "Compliant",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "isEnabled": true,
+          "retentionDays": 15
+        }
+      }
+    },
+    {
+      "name": "Compliant",
+      "type": "Microsoft.Network/firewallPolicies",
+      "apiVersion": "2022-07-01",
+      "properties": {
+        "insights": {
+          "isEnabled": true,
+          "retentionDays": 0
+        }
+      }
+    }
```

Comment:
You can also add a cases where a `type` is different, `isEnabled` is not a boolean literal, `retentionDays` is not numeric literal

Reply:
Should already be inside there. I will have a look.